stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -32,6 +26,14 @@ _DistFinderType: TypeAlias = Callable[[_T, str, bool], Iterable[Distribution]]
 _NSHandlerType: TypeAlias = Callable[[_T, str, str, types.ModuleType], str | None]
 _ResourceStream: TypeAlias = Incomplete  # A readable file-like object
 
+# TODO: Use _typeshed.importlib.LoaderProtocol after mypy 1.11 is released
+class _LoaderProtocol(Protocol):
+    def load_module(self, fullname: str, /) -> types.ModuleType: ...
+
+# typeshed only
+_D = TypeVar("_D", bound=Distribution)
+_StrictInstallerType: TypeAlias = Callable[[Requirement], _D]
```

Comment:
I don't wanna cause conflicts with your PR so I'll do it as a follow-up. Just as a note that this is added as
```suggestion
_DistributionT = TypeVar("_DistributionT", bound="Distribution")
_InstallerTypeT: TypeAlias = Callable[["Requirement"], "_DistributionT"]
```
in the next version.

https://github.com/pypa/setuptools/pull/4390/files#diff-e3d3d454fa3a072c9f46f8affa27513892fbc2d245e87f57a5927a7be851de05R117